lib/sprockets/asset.rb

```diff
@@ -64,6 +64,8 @@ def to_hash
     #
     # Returns String.
     def digest_path
+      etag = DigestUtils.pack_hexdigest(version + digest)
```

Comment:
The `etag` method should also change value since we want to bust the cache.

Reply:
Unless I'm mistaken, the cache key used doesn't depend on the `etag` method. The cache gets correctly busted when the env version changes.
There is a test for this https://github.com/rails/sprockets/blob/9f4b2f5fb77257b789c920f72dedbf0a1c0cafd6/test/test_caching.rb#L87-L98


The `etag` method needs to remain the result of computing the source, since it is used for the asset integrity. If we modify that method to return a hash to compute the environment version and the source, we are going to end up with the same issue

Reply:
`etag` is for browser cache, so. changing its value it will force the browser cache to also be invalidated which is what we want with changing the source. I know that if the version changes the filename changes and the browser cache as well, but so should the `etag`. If `etag` is being used to integrity we should stop doing that and use the hexdigest.

Reply:
Sorry, you are right `etag` is not used for integrity I got confused. Integrity uses the `digest`. We aren't changing the digest so we are good https://github.com/rails/sprockets-rails/blob/c269f5e01fdffa5c41350e855183d94ff33d318a/lib/sprockets/rails/helper.rb#L344

I'll make the change to modify the etag as well